engine/errors.go

```diff
@@ -158,3 +158,32 @@ func LogErrorWithMsg(log zerolog.Logger, msg string, err error) {
 	// all other errors should just be logged as usual
 	log.Error().Str("error_type", "internal_error").Err(err).Msg(msg)
 }
+
+// UnknownExecutionResultError indicates that the Execution Result is unknown
+type UnknownExecutionResultError struct {
```

Comment:
This seems like a very specific error condition and is not thrown from an engine - not sure it belongs as a general `engine` package error. We could put this with the definition the consensus mempool where the error is thrown.